zenodo/modules/pages/utils.py

```diff
@@ -81,9 +112,70 @@ def send_support_email(context):
         body=msg_body
     )
 
-    if context['form'].attachments.data:
-        msg.attach(context['form'].attachments.data.filename,
-                   'application/octet-stream',
-                   context['form'].attachments.data.read())
+    attachments = request.files.getlist("attachments")
+    if attachments:
+        if check_attachment_size(attachments):
+            for upload in attachments:
+                msg.attach(upload.filename,
+                           'application/octet-stream',
+                           upload.read())
+        else:
+            return "error"
 
     mail.send(msg)
+
+
+def user_agent_information():
+    """Function for user agent information."""
+    uap = user_agent_parser.Parse(str(request.user_agent))
+
+    # Operating System and it's version.
+    os = ''
+    if uap['os']['family'] is not None:
+        os = uap['os']['family']
+        if uap['os']['major'] is not None:
+            os += ' (' + str(uap['os']['major'])
+            if uap['os']['patch_minor'] is not None:
+                os += '.' + str(uap['os']['patch_minor'])
+            if uap['os']['minor'] is not None:
+                    os += '.' + str(uap['os']['minor'])
+            if uap['os']['patch'] is not None:
+                os += '.' + str(uap['os']['patch'])
+            os += ')'
+
+    # Browser and it's version.
+    browser = ''
+    if uap['user_agent']['family'] is not None:
+        browser = uap['user_agent']['family']
+        if uap['user_agent']['major'] is not None:
+            browser += ' (' + str(uap['user_agent']['major'])
+            if uap['user_agent']['major'] is not None:
+                browser += '.' + str(uap['user_agent']['minor'])
+            if uap['user_agent']['major'] is not None:
+                browser += '.' + str(uap['user_agent']['patch'])
+            browser += ')'
+
+    # Device, it's model and brand.
+    device = ''
+    if uap['device']['family'] is not None:
+        device = uap['device']['family']
+        if uap['device']['brand'] is not None:
+            device += ' ' + uap['device']['brand']
+        if uap['device']['model'] is not None:
+            device += ' ' + uap['device']['model']
+
+    return os, browser, device
+
+
+def display_size(size):
```

Comment:
You can also use the [`do_filesizeformat` Jinja function](https://github.com/pallets/jinja/blob/d3ed6c7947172fe96342a38c5fd779f205cf1527/jinja2/filters.py#L371-L398)